### ISSUE TITLE
Remove componentWillReceiveProps, React 16.9.0 deprecates it.

### ISSUE THREAD

**Issue by jimnanney** [moroshko/react-autowhatever#50]
React 16.9.0 deprecates the componentWillReceiveProps lifecycle method and throws warnings for it use.

Looking through Autowhatever.js, it appears it is only used for computing theme, sectionItems and sectionIterator. These could be moved to the render function, and if it is expensive, could be memoized.

**Comment by netraameht** [comment]
Just renaming to UNSAFE_componentWillReceiveProps will break older React versions.

**Comment by jimnanney** [comment]
My PR did not rename it, just moved the functionality into the render function.

**Comment by netraameht** [comment]
That’s right, however I’m commenting because commit 652a9090 did renamed the function.

**Comment by netraameht** [comment]
That’s right, however commit 652a9090 did rename the function.
